src/server/debug.rs

```diff
@@ -970,13 +970,21 @@ impl MvccChecker {
             // If lock exists, check whether the records in DEFAULT and WRITE
             // match it.
             if let Some(ref l) = lock {
-                // All write records' ts should be less than lock's ts.
+                // All write records's ts should be less than the for_update_ts (if the
+                // lock is a pessimistic lock) or the ts of the lock.
```

Comment:
I think when a pessimistic lock is replaced with a normal lock, its start_ts will be kept. So when pessimistic transaction is enabled, the non-pessimistic locks can also have less ts than the latest write does.

Reply:
Oh, yes. Should use `max(l.ts, l.for_update_ts)`.